include/llbuild/BuildSystem/BuildDescription.h

```diff
@@ -149,6 +149,34 @@ class Node {
       ArrayRef<std::pair<StringRef, StringRef>> values) = 0;
 };
 
+/// Definition for a file system stat used by the build file.
+class Stat {
```

Comment:
I'm not sure why we would need this in the build description layer. I was expecting this to just be a node created transparently by the filtering logic, but not exposed in the BuildDescription.

Reply:
To be more explicit, it would have a custom task implementation, and need a BuildKey entry, but it shouldn't surface in the BuildDescription.